### PR TITLE
Disease protection

### PR DESCRIPTION
**Pull request by kiVts**
# About the pull request
Wearing mask/latex gloves/bio suit/space suit will now impact the likeliness of you getting infected from flu/cold/whatever. Doesnt affect black goo(zombie)

I also have absolutely no idea what Astar proc is and why its there so I replaced that with adjacent. UPD: astar returned. its pathfinding.

Nuked one single letter var so be happy

# Explain why it's good for the game
More interesting moments with those events of flu disease once in a decade, like mask and gloves distribution.(perhaps we will have more of them now)
Plus I was so sad that wearing new cool biosuit doesn't protect you from it.
Wearing full kit wont give 100% immunity, final random roll always does -5
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: Kivts
add: You can partially protect yourself from flu and other disease by wearing a mask and other PPE. Doesnt affect Black Goo.
/:cl:

### CODE REVIEW

File: code/datums/disease.dm [hunk 1]
```diff
@@ -173,6 +174,21 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
 		var/mob/living/carbon/human/H = affected_mob
 		H.med_hud_set_status()
 
+/datum/disease/proc/get_infection_chance(mob/living/carbon/human/victim)
+	if(ishuman(victim))
+		var/protection = 5 //Our current credit protection on how good are we protected from any sort of disease. Five acts as base protection.
```
Comment (BeagleGaming1) [inline code/datums/disease.dm:179]: Why start protection at 5 if you will decrease it by 5 in the end?

Comment (kiVts) [reply code/datums/disease.dm:179]: I clamp it anyways, you are right

File: code/datums/disease.dm [hunk 2]
```diff
@@ -121,10 +121,11 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
...
 	if(isturf(source.loc))
-		for(var/mob/living/carbon/M in oview(check_range, source))
-			if(isturf(M.loc))
-				if(AStar(source.loc, M.loc, /turf/proc/AdjacentTurfs, /turf/proc/Distance, check_range))
-					M.contract_disease(src, 0, 1, force_spread)
+		for(var/mob/living/carbon/victim in oview(check_range, source))
+			if(isturf(victim.loc))
+				if(AStar(source.loc, victim.loc, /turf/proc/AdjacentTurfs, /turf/proc/Distance, check_range) && !locate(src) in victim.viruses)
+					if(get_infection_chance(victim))
+						victim.contract_disease(src, 0, 1, force_spread)
```
Comment (harryob) [inline code/datums/disease.dm:128]: contract disease should be checking for all this in line 81

[passed = check_disease_pass_clothes(target_zone)](https://github.com/cmss13-devs/cmss13/blob/ab44c0dd2a9ea424cb7585d20d5d0905ee9a537d/code/datums/diseases/mob_procs.dm#L81)

and it does it a little neater with a variable on clothes, is that not working correctly?

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/datums/disease.dm (hunk 2 above)
Comment (kiVts) [reply code/datums/disease.dm:128]: You get infected almost instantly without these bits of code, but I will double check.

Comment (kiVts) [reply code/datums/disease.dm:128]: I guess I can nuke my changes and beat this to work.

**Comment by kiVts** [comment]
No idea why did I do this. I will fix it soon:tm:

**Comment by kiVts** [comment]
nevermind, round two. it doesnt work with the old code. leaving as is.

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**